crates/typed-store/src/rocks/mod.rs

```diff
@@ -578,6 +709,45 @@ impl DBBatch {
     }
 }
 
+pub struct DBTransaction<'a> {
+    rocksdb: Arc<RocksDB>,
+    transaction: Transaction<'a, rocksdb::OptimisticTransactionDB>,
+}
+
+impl<'a> DBTransaction<'a> {
+    pub fn new(db: &'a Arc<RocksDB>) -> Result<Self, TypedStoreError> {
+        Ok(Self {
+            rocksdb: db.clone(),
+            transaction: db.transaction()?,
+        })
+    }
+
+    pub fn insert_batch<J: Borrow<K>, K: Serialize, U: Borrow<V>, V: Serialize>(
+        self,
+        db: &DBMap<K, V>,
+        new_vals: impl IntoIterator<Item = (J, U)>,
+    ) -> Result<Self, TypedStoreError> {
+        if !Arc::ptr_eq(&db.rocksdb, &self.rocksdb) {
+            return Err(TypedStoreError::CrossDBBatch);
+        }
+
+        new_vals
+            .into_iter()
+            .try_for_each::<_, Result<_, TypedStoreError>>(|(k, v)| {
+                let k_buf = be_fix_int_ser(k.borrow())?;
+                let v_buf = bincode::serialize(v.borrow())?;
+                self.transaction.put_cf(&db.cf(), k_buf, v_buf)?;
+                Ok(())
+            })?;
+        Ok(self)
+    }
+
+    pub fn commit(self) -> Result<(), TypedStoreError> {
+        self.transaction.commit()?;
```

Comment:
We also need to have `get_for_update` and `snapshot` for optimistic transactions. I think rust_rocksdb has these iirc